docs/instrumentation_examples.md

```diff
@@ -0,0 +1,181 @@
+```python
+from armory.instrument import get_hub, Meter
+
+def set_up_meter():
+    meter = Meter(
+        "my_arbitrary_meter_name", lambda x: x, "my_attack.attack_output"
+    )
+    get_hub().connect_meter(meter)
+```
+#### Step 3
+As before, the `Meter` accepts 3 inputs: a meter name, a metric/function for processing, and a argument name to pass the metric/function.
+- The meter name (`"my_arbitrary_meter_name"`) can be arbitrary within this context
+- Again, `Meter` will record variables monitored by `Probe` as-is (thus `lambda x: x`)
+- The argument passed to the metric/function follows a `.`-separated format (`"my_attack.attack_output"`), which needs to be consistent with `Probe` setup earlier:
+    - `my_attack` matches input in `probe = get_probe("my_attack")`
+    - `attack_output` matches variable name in `probe.update(attack_output=x_patch)`
+
+#### Step 4
+Again, `get_hub().connect_meter(meter)` is necessary for establishing the connection between `meter` created in `armory/user_init_script.py` and `probe` created in `armory/custom_attack.py`.
+
+### Config Setup
+Last but not least, the config file passed to `armory run` needs to be updated for these changes to take effect, which is accomplished by adding the `"user_init"` block (please refer to [User Initialization](./scenarios.md#user-initialization) for more details):
+```json
+...
+    "user_init": {
+        "module": "user_init_script",
+        "name": "set_up_meter"
+    },
+...
+```
+This will prompt armory to run `set_up_meter` in `user_init_script.py` before anything else is loaded for the scenario.
+
+## Saving Results
+By default, outputs from `Meter`s will be saved to the output `json` file after `armory run`. Whether this suffices for the user depends on what the user is trying to measure.
+
+Users who have tried the examples in this document, however, may run into some of the following warning logs:
+> 2022-12-16 19:34:36 30s WARNING  armory.instrument.instrument:_write:856 record (name=my_arbitrary_meter_name, batch=0, result=...) size > max_record_size 1048576. Dropping.
+
+Outputs are saved to a `json` file because of a default `ResultWriter` class tied to the `Meter` class, which has a `max_record_size` limit for each record. Any record that exceeds `max_record_size` will not save to the `json` file. That the outputs exceed a size limit also suggests that a `json` file may not be the best file type to save to. To work around these behaviors, we can define a new `Writer` subclass (`ResultWriter` is also a `Writer` subclass) to work with our examples that does not have a size limit and will save to another filetype, such as a `png` file, since we are saving data for an image. Below is an updated `user_init_script.py` for Example 2 with a new `ImageWriter` class, which uses the `export` method of `ImageClassificationExporter` to save an image, and a `set_up_meter_writer` function that will be executed with the `user_init` block:
```

Comment:
one super minor comments: we changed the code to `ObjectDetectionExporter`, but the text still says `ImageClassificationExporter`. We can merge once this change is made, no need to wait for CI again

Reply:
resolved with 6de62f2